src/compiler/transformers/decorators-to-static/decorator-utils.ts

```diff
@@ -1,20 +1,35 @@
 import ts from 'typescript';
 
 import { objectLiteralToObjectMap } from '../transform-utils';
-import { StencilDecorator } from './decorators-constants';
+import type { StencilDecorator } from './decorators-constants';
 
-export const getDeclarationParameters: GetDeclarationParameters = (decorator: ts.Decorator): any => {
+export const getDecoratorParameters: GetDecoratorParameters = (
+  decorator: ts.Decorator,
+  typeChecker: ts.TypeChecker,
+): any => {
   if (!ts.isCallExpression(decorator.expression)) {
     return [];
   }
-  return decorator.expression.arguments.map(getDeclarationParameter);
+  return decorator.expression.arguments.map((arg) => getDecoratorParameter(arg, typeChecker));
 };
 
-const getDeclarationParameter = (arg: ts.Expression): any => {
+const getDecoratorParameter = (arg: ts.Expression, typeChecker: ts.TypeChecker): any => {
   if (ts.isObjectLiteralExpression(arg)) {
     return objectLiteralToObjectMap(arg);
   } else if (ts.isStringLiteral(arg)) {
     return arg.text;
+  } else if (ts.isPropertyAccessExpression(arg) || ts.isIdentifier(arg)) {
+    const type = typeChecker.getTypeAtLocation(arg);
+    if (type !== undefined && type.isLiteral()) {
+      /**
+       * Enum members are property access expressions, so we can evaluate them
+       * to get the enum member value as a string.
+       *
+       * This enables developers to use enum members in decorators.
+       * e.g. @Watch(MyEnum.VALUE)
+       */
```

Comment:
We may wanna tweak this comment a bit - I think the scope of this has changed quite a bit now (since we support all property expr + identifiers that are literals for decorators other than `@Watch`)